File: app/src/main/java/com/google/sample/cloudvision/searchpojos/Image.java

package com.google.sample.cloudvision.searchpojos;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Image implements Parcelable
{

    @SerializedName("contextLink")
    @Expose
    private String contextLink;
    @SerializedName("height")
    @Expose
    private Integer height;
    @SerializedName("width")
    @Expose
    private Integer width;
    @SerializedName("byteSize")
    @Expose
    private Integer byteSize;
    @SerializedName("thumbnailLink")
    @Expose
    private String thumbnailLink;
    @SerializedName("thumbnailHeight")
    @Expose
    private Integer thumbnailHeight;
    @SerializedName("thumbnailWidth")
    @Expose
    private Integer thumbnailWidth;
    public final static Creator<Image> CREATOR = new Creator<Image>() {


        @SuppressWarnings({
            "unchecked"
        })
        public Image createFromParcel(Parcel in) {
            Image instance = new Image();
            instance.contextLink = ((String) in.readValue((String.class.getClassLoader())));
            instance.height = ((Integer) in.readValue((Integer.class.getClassLoader())));
            instance.width = ((Integer) in.readValue((Integer.class.getClassLoader())));
            instance.byteSize = ((Integer) in.readValue((Integer.class.getClassLoader())));
            instance.thumbnailLink = ((String) in.readValue((String.class.getClassLoader())));
            instance.thumbnailHeight = ((Integer) in.readValue((Integer.class.getClassLoader())));
            instance.thumbnailWidth = ((Integer) in.readValue((Integer.class.getClassLoader())));
            return instance;
        }

        public Image[] newArray(int size) {
            return (new Image[size]);
        }

    }
    ;

    public String getContextLink() {
        return contextLink;
    }

    public void setContextLink(String contextLink) {
        this.contextLink = contextLink;
    }

    public Integer getHeight() {
        return height;
    }

    public void setHeight(Integer height) {
        this.height = height;
    }

    public Integer getWidth() {
        return width;
    }

    public void setWidth(Integer width) {
        this.width = width;
    }

    public Integer getByteSize() {
        return byteSize;
    }

    public void setByteSize(Integer byteSize) {
        this.byteSize = byteSize;
    }

    public String getThumbnailLink() {
        return thumbnailLink;
    }

    public void setThumbnailLink(String thumbnailLink) {
        this.thumbnailLink = thumbnailLink;
    }

    public Integer getThumbnailHeight() {
        return thumbnailHeight;
    }

    public void setThumbnailHeight(Integer thumbnailHeight) {
        this.thumbnailHeight = thumbnailHeight;
    }

    public Integer getThumbnailWidth() {
        return thumbnailWidth;
    }

    public void setThumbnailWidth(Integer thumbnailWidth) {
        this.thumbnailWidth = thumbnailWidth;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(contextLink);
        dest.writeValue(height);
        dest.writeValue(width);
        dest.writeValue(byteSize);
        dest.writeValue(thumbnailLink);
        dest.writeValue(thumbnailHeight);
        dest.writeValue(thumbnailWidth);
    }

    public int describeContents() {
        return  0;
    }

}

File: app/src/main/java/com/google/sample/cloudvision/searchpojos/Image.kt
package com.google.sample.cloudvision.searchpojos

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Image : Parcelable {

    @SerializedName("contextLink")
    @Expose
    var contextLink: String? = null
    @SerializedName("height")
    @Expose
    var height: Int? = null
    @SerializedName("width")
    @Expose
    var width: Int? = null
    @SerializedName("byteSize")
    @Expose
    var byteSize: Int? = null
    @SerializedName("thumbnailLink")
    @Expose
    var thumbnailLink: String? = null
    @SerializedName("thumbnailHeight")
    @Expose
    var thumbnailHeight: Int? = null
    @SerializedName("thumbnailWidth")
    @Expose
    var thumbnailWidth: Int? = null

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(contextLink)
        dest.writeValue(height)
        dest.writeValue(width)
        dest.writeValue(byteSize)
        dest.writeValue(thumbnailLink)
        dest.writeValue(thumbnailHeight)
        dest.writeValue(thumbnailWidth)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        val CREATOR: Parcelable.Creator<Image> = object : Parcelable.Creator<Image> {


            override fun createFromParcel(`in`: Parcel): Image {
                val instance = Image()
                instance.contextLink = `in`.readValue(String::class.java.classLoader) as String
                instance.height = `in`.readValue(Int::class.java.classLoader) as Int
                instance.width = `in`.readValue(Int::class.java.classLoader) as Int
                instance.byteSize = `in`.readValue(Int::class.java.classLoader) as Int
                instance.thumbnailLink = `in`.readValue(String::class.java.classLoader) as String
                instance.thumbnailHeight = `in`.readValue(Int::class.java.classLoader) as Int
                instance.thumbnailWidth = `in`.readValue(Int::class.java.classLoader) as Int
                return instance
            }

            override fun newArray(size: Int): Array<Image?> {
                return arrayOfNulls(size)
            }

        }
    }

}
